src/test/regress/expected/multi_truncate.out

```diff
@@ -364,5 +374,132 @@ SELECT * FROM test_local_truncate;
  1 | 2
 (1 row)
 
+DROP TABLE test_local_truncate;
+-- Test truncate_local_table UDF
+CREATE TABLE referenced_table(id int UNIQUE, test_column int);
+INSERT INTO referenced_table SELECT x,x FROM generate_series(1,10000) x;
+CREATE TABLE referencing_table(id int, ref_id int REFERENCES referenced_table(id));
+INSERT INTO referencing_table SELECT * FROM referenced_table;
+-- The following will fail as the table is not distributed
+SELECT truncate_local_table('referenced_table');
+ERROR:  cannot execute truncate_local_table on given relation
+HINT:  This UDF only truncates local records of distributed tables.
+-- Test foreign keys from local tables to distributed tables
+-- We can not truncate local tables until all the local foreign keys are removed.
+SELECT create_distributed_table('referenced_table', 'id');
```

Comment:
Would be good to have some tests for create_reference_table as well.

Reply:
added some cases below